packages/contracts/contracts/identity/IdentityProxy.s

```diff
@@ -0,0 +1,200 @@
+pragma solidity ^0.4.24;
+
+interface ERC725 {
+    event DataChanged(bytes32 indexed key, bytes32 indexed value);
+    event OwnerChanged(address indexed ownerAddress);
+    event ContractCreated(address indexed contractAddress);
+
+    function changeOwner(address _owner) external;
+    function getData(bytes32 _key) external view returns (bytes32 _value);
+    function setData(bytes32 _key, bytes32 _value) external;
+    function execute(uint256 _operationType, address _to, uint256 _value, bytes _data) external payable;
+}
+
+interface ERC20 {
+    function approve(address _spender, uint256 _value) public;
+}
+
+contract IdentityProxy is ERC725 {
+
+    uint256 constant OPERATION_CALL = 0;
+    uint256 constant OPERATION_CREATE = 1;
+
+    mapping(bytes32 => bytes32) store;
+
+    mapping(address => uint) public nonce;
+
+    // the owner
+    address public owner;
+
+    event Forwarded (bytes sig, address signer, address destination, uint value, bytes data, bytes32 _hash);
+
+    modifier onlyOwner() {
+        require(owner == 0x0 || msg.sender == owner, "only-owner-allowed");
+        _;
+    }
+
+    function () external payable {}
+
+    function changeOwner(address _owner)
+        external
+        onlyOwner
+    {
+        owner = _owner;
+        emit OwnerChanged(owner);
+    }
+
+    function getData(bytes32 _key)
+        external
+        view
+        returns (bytes32 _value)
+    {
+        return store[_key];
+    }
+
+    function setData(bytes32 _key, bytes32 _value)
+        external
+        onlyOwner
+    {
+        store[_key] = _value;
+        emit DataChanged(_key, _value);
+    }
+
+    function execute(uint256 _operationType, address _to, uint256 _value, bytes _data)
+        external
+        payable
+        onlyOwner
+    {
+        if (_operationType == OPERATION_CALL) {
+            executeCall(_to, _value, _data);
+        } else if (_operationType == OPERATION_CREATE) {
+            address newContract = executeCreate(_data);
+            emit ContractCreated(newContract);
+        } else {
+            // We don't want to spend users gas if parametar is wrong
+            revert("invalid operationType");
+        }
+    }
+
+    // copied from GnosisSafe
+    // https://github.com/gnosis/safe-contracts/blob/v0.0.2-alpha/contracts/base/Executor.sol
+    function executeCall(address to, uint256 value, bytes memory data)
+        internal
+        returns (bool success)
+    {
+        // uint256 txGas = 1 ether;
+        // solium-disable-next-line security/no-inline-assembly
+        assembly {
+            success := call(gas, to, value, add(data, 0x20), mload(data), 0, 0)
+        }
+    }
+
+    // copied from GnosisSafe
+    // https://github.com/gnosis/safe-contracts/blob/v0.0.2-alpha/contracts/base/Executor.sol
+    function executeCreate(bytes memory data)
```

Comment:
Do we need this method? We are not deploying contracts with Proxy right now. Will it be needed in future? Unless we want the user's proxy to be not specific to Origin, I guess we can remove contract creation from this file.

Reply:
We're using the ERC725 v2 standard for the account... though that could change later